workflow/lib/preprocess/file_utils.py

```diff
@@ -40,13 +36,35 @@ def get_sample_fps(
     if cycle is not None:
         filtered_df = filtered_df[filtered_df["cycle"] == int(cycle)]
 
-    # If we have a channel column, return list of files in specified order
-    if "channel" in filtered_df.columns:
-        # Create dictionary mapping channel to file path
-        channel_to_file = dict(zip(filtered_df["channel"], filtered_df["sample_fp"]))
+    if round_order is not None:
+        # Filter to only include specified rounds
+        filtered_df = filtered_df[filtered_df["round"].isin(round_order)]
+        # Create dictionary mapping round to DataFrame rows
+        round_groups = {round_num: group for round_num, group in filtered_df.groupby("round")}
+        
+        # Initialize list to store file paths for each round
+        round_files = []
+        
+        # Process each round in the specified order
+        for round_num in round_order:
+            if round_num not in round_groups:
+                raise ValueError(f"Round {round_num} not found in data")
+                
+            round_df = round_groups[round_num]
+            
+            # If we have channels and channel order is specified
+            if "channel" in round_df.columns and channel_order is not None:
+                channel_to_file = dict(zip(round_df["channel"], round_df["sample_fp"]))
+                round_files.extend([channel_to_file[channel] for channel in channel_order])
+            else:
+                round_files.append(round_df["sample_fp"].iloc[0])
+        
+        return round_files
 
-        # Return files in specified order
+    # If no rounds specified but we have channels and channel order
+    if "channel" in filtered_df.columns and channel_order is not None:
+        channel_to_file = dict(zip(filtered_df["channel"], filtered_df["sample_fp"]))
         return [channel_to_file[channel] for channel in channel_order]
 
-    # Otherwise return single file path (original behavior)
-    return filtered_df["sample_fp"].iloc[0]
+    # Otherwise return single file path
+    return filtered_df["sample_fp"].iloc[0]
```

Comment:
```suggestion
    return filtered_df["sample_fp"].iloc[0]

```

Missing newline at EOF. But more importantly this means that a check with Ruff was missed. Could you check the code with `ruff check` and `ruff format`? I will add a note to do this on the PR template to remind people.